proto/cdcpb.proto

```diff
@@ -166,10 +174,17 @@ message ChangeDataRequest {
     }
 
     KvAPI kv_api = 11;
+
     // Whether to filter out the value write by cdc itself.
     bool filter_loop = 12;
 }
 
 service ChangeData {
     rpc EventFeed(stream ChangeDataRequest) returns(stream ChangeDataEvent);
+
+    // EventFeedV2 is like EventFeed, with some new changes:
+    //  * multiplexing, which means one region can be subscribed multiple times
+    //    with different `request_id` in one RPC stream. And, it can be
+    //    deregistered with a `Deregister` message with a given `request_id`.
+    rpc EventFeedV2(stream ChangeDataRequest) returns (stream  ChangeDataEvent);
```

Comment:
The RPC signature is identical to `EventFeed`,  can we extend `EventFeed` instead?

TiKV CDC component has feature gate which we can enable or disable multiplexing.

Reply:
Reuse `EventFeed` can lead mistakes if TiCDC version is higher than TiKV.
So `EventFeedV2` is required for this case. And I have added version and requested feature checks in `EventFeedV2`.
Please check [here](https://github.com/tikv/tikv/pull/14923/files#diff-ae1198b4c32440ad526f4bfc1ea4ce851d7d46aa9ba8b9000c37a42d770b64b4R411).

Reply:
/cc @hi-rustin @asddongmen

Reply:
> Reuse EventFeed can lead mistakes if TiCDC version is higher than TiKV.

Why not let TiCDC check TiKV version before sending requests.

Reply:
If you mean we ask this infomation from PD, I think it's more complex than using another RPC. We must handle this case: PD tells us `tikv is v 6.6.6` but when establishing the connection, tikv has been restarted in `v6.6.7`.

Reply:
> If you mean we ask this infomation from PD, I think it's more complex than using another RPC. 

Checking PD is like 5-6 lines of code, but making a new RPC it duplicates at least 100 lines of code (see event_feed implementation in tikv).
I don't think the new RPC is necessary, given the fact that they has the same signature and it adds extra maintenance burden.

> PD tells us tikv is v 6.6.6 but when establishing the connection, tikv has been restarted in v6.6.7

It's rare case, because TiCDC is upgraded after TiKV. And even if it happens it won't cause issue, because new tikv must be compatible with old protocol.

Reply:
The behavior will be changed soon. TiCDC will be upgraded before TiKV later.

Reply:
It's still the rare case, because tikv updates version in pd before starts serving, very little chance of triggering such issue, and even if it happens it won't cause issue, because new tikv must be compatible with old protocol.

Also if you really care about this rare issue, you can probe tikv by sending two requests for DDL region (which is needed for changefeeds that have more than 2 tables).